nixos/modules/system/boot/systemd-lib.nix

```diff
@@ -147,7 +147,13 @@ in rec {
       done
 
       # Symlink all units provided listed in systemd.packages.
-      for i in ${toString cfg.packages}; do
+      packages="${toString cfg.packages}"
```

Comment:
Would not `toString (lib.unique cfg.packages)` work?

Reply:
`lib.unique` is O(n^2), though not sure if that will realistically cause problems. Otherwise I'd say applying unique to `systemd.packages` would be close to ideal (having a built in set type would be the best though :).

Reply:
Wouldn't a built in set type be `attrsOf`? I think it would be un-intuitive for `systemd.packages` to have to be declared in anything other than a list, and because of `systemPackages`.
On the topic of being O(n^2), there's a deprecated type that is basically this suggestion `uniqList`.

When it comes to removing duplicates in bash versus `lib.unique`,  idk.

Reply:
> Wouldn't a built in set type be attrsOf?

Right, I was thinking more like a math set, ie. something like `uniqList`. Agreed that it makes the most UX sense for it to be list.

> When it comes to removing duplicates in bash versus lib.unique, idk.

Yep, I'm fine with both. Doing it through the option's apply with `unique` is clean and will guard against any other uses, but O(n^2) isn't ideal so :man_shrugging: